data.php

```diff
@@ -156,11 +156,12 @@ function readInBlockList() {
         $file="/etc/pihole/gravity.list";
         $linecount = 0;
         $handle = fopen($file, "r");
-        while(!feof($handle)){
+        if(gettype($handle) == "resource") {
```

Comment:
The fopen code is fine, but it would actually be faster to call `wc -l` on the file through an exec().

I am more concerned about the efficiency of calling gettype, especially for resources when a simple boolean check would do